test/functional/page_objects/settings_page.js

```diff
@@ -510,13 +510,12 @@ export function SettingsPageProvider({ getService, getPageObjects }) {
 
     async setScriptedFieldScript(script) {
       log.debug('set scripted field script = ' + script);
-      const field = await testSubjects.find('editorFieldScript');
-      const currentValue = await field.getAttribute('value');
-      if (script === currentValue) {
-        return;
+      const aceEditorCssSelector = '[data-test-subj="codeEditorContainer"] .ace_editor';
+      await find.clickByCssSelector(aceEditorCssSelector);
+      for (let i = 0; i < 1000; i++) {
+        await browser.pressKeys(browser.keys.BACK_SPACE);
       }
-      await field.clearValueWithKeyboard({ charByChar: true });
-      await field.type(script);
+      await browser.pressKeys(...script.split(''));
```

Comment:
naive question - I wonder if this could be simplified by using the return value from `find.byCssSelector`, and using that to clear/set the field value?
```javascript
      const aceEditorCssSelector = '[data-test-subj="codeEditorContainer"] .ace_editor';
      const input = await find.byCssSelector(aceEditorCssSelector);
      await input.clearValue();
      await input.type(script);
```
I haven't tested this -- and I'm not certain if it will work the same way with a `textarea`, but just thought I'd ask if you'd tried that yet, as it is simpler to read. This is similar to how we've implemented the `setValue` method in visualize tests: https://github.com/elastic/kibana/blob/f5bbe820230aeaad6186f2c1a5b2ed88b24d44e1/test/functional/page_objects/visualize_page.js#L415-L421

Reply:
++

Reply:
The thing is, it is not a `<textarea>`, it is Ace Editor. It actually has `<textarea>` inside, but it is only the size of one char and invisible and it does not have any value:

![image](https://user-images.githubusercontent.com/9773803/57321141-f40de200-7100-11e9-876c-b2c80c5c2fde.png)

As I understand, the `<textarea>` is there only for the caret, and possibly to listen for key presses. But the actual "value" is rendered in HTML:

![image](https://user-images.githubusercontent.com/9773803/57321305-5a930000-7101-11e9-9dcb-bd78bc8900b1.png)

`clearValue()` does

![image](https://user-images.githubusercontent.com/9773803/57321959-b90cae00-7102-11e9-8881-2eb0678b30e0.png)

which shouldn't work with Ace Editor, I think.

Reply:
BTW, there is `input.clearValueWithKeyboard()`, but it did not work for, not sure why.

Reply:
Ah interesting, I didn’t know that. Thanks for the explanation!